mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -781,4 +808,117 @@ class LogisticRegressionSuite extends SparkFunSuite with MLlibTestSparkContext {
         .forall(x => x(0) >= x(1)))
 
   }
+
+  test("binary logistic regression with weighted samples") {
```

Comment:
This will be hard since it's not easy to reproduce the same dataset in R given different random generator implementation. But I don't think we need to compare with R's glmnet here since we are comparing against with our spark glmnet implementation without weighted feature, and that has been verified against R.